src/mbed_os_tools/test/host_tests_plugins/module_copy_jlink.py

```diff
@@ -0,0 +1,113 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+
+from .host_test_plugins import HostTestPluginBase
+
+
+COPY_FILE = "copy.jlink"
+
+class HostTestPluginCopyMethod_Jlink(HostTestPluginBase):
+
+    # Plugin interface
+    name = 'HostTestPluginCopyMethod_Jlink'
+    type = 'CopyMethod'
+    capabilities = ['JTAG', 'SWD']
+    required_parameters = ['image_path', 'mcu']
+
+    def is_os_supported(self, os_name=None):
+        """! In this implementation this plugin only is supporeted under Windows machines
+        """
+        # If no OS name provided use host OS name
+        if not os_name:
+            os_name = self.mbed_os_support()
+
+        # This plugin only works on Windows
+        if os_name and os_name.startswith('Windows'):
+            return True
+        return False
+
+    def setup(self, *args, **kwargs):
+        """! Configure plugin, this function should be called before plugin execute() method is used.
+        """
+        #Note you need to have jlink.exe on your system path!
+        self.JLINK = 'JLink.exe'
+        return True
+    
+    def build_jlink_script(self, path, image_path):
+        """! write a jlink commander file that
+            erase and flash the target
+
+        @param path the file path of the script we are writing in the function
+        @param image_path path to the binary file to flash
+        """
+        with open(path, "w") as jlink_script:
+            jlink_script.write("connect\n")
+            jlink_script.write("erase\n")
+            jlink_script.write("loadbin {0} 0x08000000\n".format(image_path))
+            jlink_script.write("r\n")
+            jlink_script.write("go\n")
+            jlink_script.write("exit\n")
+
+
+    def execute(self, capability, *args, **kwargs):
+        """! Executes capability by name
+
+        @param capability Capability name
+        @param args Additional arguments
+        @param kwargs Additional arguments
+
+        @details Each capability e.g. may directly just call some command line program or execute building pythonic function
+
+        @return Capability call return value
+        """
+        result = False
+        if self.check_parameters(capability, *args, **kwargs) is True:
+            
+            mcu = os.path.normpath(kwargs['mcu'])
+            image_path = os.path.normpath(kwargs['image_path'])
+
+            jlink_file_path = os.path.join(tempfile.gettempdir(), COPY_FILE)
```

Comment:
If there are multiple processes running at once, I believe this it is possible for multiple processes trying to write and read to the same file. I think an alternative to this would be to use [`tempfile.mkstemp()`](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp) instead.

Reply:
Sure. We used the same `tempfile` methods in the ST-LINK plugin bugfix. Should I open a separate PR for changing to the STLINK plugin to use `mkstemp` as well?

Reply:
That would be great, thanks!